.github/workflows/test.yml

```diff
@@ -1,44 +1,28 @@
-name: Test
+name: Lint and test
 
 on:
+  pull_request:
+    branches:
+    - cairo-1
   push:
-    tags:
-    - "v*"
+    branches:
+    - cairo-1
 
 jobs:
-  validate:
+  lint_and_test:
+    name: Lint and test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - name: Get commits
-      id: commits
-      run: |
-        echo "MAIN=$(git show -s --format="%H" origin/main)" >> $GITHUB_OUTPUT
-        echo "TAG=$(git rev-list -n 1 ${GITHUB_REF#refs/*/})" >> $GITHUB_OUTPUT
-    - name: Print commits
-      run: |
-        echo "Main commit: ${{ steps.commits.outputs.MAIN }}"
-        echo "Tag commit: ${{ steps.commits.outputs.TAG }}"
-    - name: Compare commits
-      if: ${{ steps.commits.outputs.MAIN != steps.commits.outputs.TAG }}
-      uses: actions/github-script@d556feaca394842dc55e4734bf3bb9f685482fa0 # v6.3.3
+    - uses: actions/checkout@v3
       with:
-        script: |
-            core.setFailed('Tagged commit does not match main')
-
-  test:
-    runs-on: ubuntu-latest
-    needs: [validate]
-    steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.8
-      uses: actions/setup-python@v2
+        submodules: recursive
+    - name: Markdown lint
+      uses: DavidAnson/markdownlint-cli2-action@5b7c9f74fec47e6b15667b2cc23c63dff11e449e # v9
```

Comment:
i don't see any problem with this but just wondering, what's the difference with the one we had before?

Reply:
oh i just now read the PR description my bad. then it's docker right?